lib/main.dart

```diff
@@ -302,32 +271,75 @@ class SplashState extends ConsumerState<Splash> with WidgetsBindingObserver {
       FlutterNativeSplash.remove();
 
       if (session.isLoggedOut) {
-        await _goToIntroScreen();
+        _goToIntroScreen();
         return;
       }
 
       Navigator.of(context).pushReplacementNamed('/home');
     } catch (e, stack) {
       dev.log(e.toString(), error: e, stackTrace: stack);
       FlutterNativeSplash.remove();
-      await _goToIntroScreen();
+      _goToIntroScreen();
+    }
+  }
+
+  Future _checkDeviceSecurity() async {
+    if (Platform.isIOS == false && Platform.isAndroid == false) {
+      return;
+    }
+
+    final preferences = await HivePreferencesDatasource.getInstance();
```

Comment:
Perhaps we could use a provider and watch it to check Device Security because if user leavexs a wallet open and changes the developer mode for example, he won't be informed

Reply:
Nice one, I hadn't thought of that ! No need to use a provider to do that I think, I will check what I can do.